app/src/util/itertools.ts

```diff
@@ -84,6 +100,46 @@ function* filter<T>(predicate: (x: T) => boolean, input: Generator<T>): Generato
   }
 }
 
+function* dropwhile<T>(predicate: (x: T) => boolean, input: Generator<T>): Generator<T> {
+  for (const item of input) {
+    if (!predicate(item)) {
```

Comment:
this seems... wrong?

Reply:
This is similar to pythons [itertools.dropwhile](https://docs.python.org/3/library/itertools.html#itertools.dropwhile). It removes elements from the beginning of the input generator as long as the predicate is met.

Reply:
ah nice, i was too banana to understand the genius of the code while reviewing :see_no_evil: